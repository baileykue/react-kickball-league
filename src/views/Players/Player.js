import { useState, useEffect } from 'react';
import { getPlayerById } from '../../services/players';
import PlayerDetail from '../../components/PlayerList/PlayerDetail';

export default function Player(props) {
  const id = props.match.params.id;
  const [player, setPlayer] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const fetchData = async () => {
      const data = await getPlayerById(id);
      setPlayer(data);
      setLoading(false);
    };
    fetchData();
  }, [id]);

  if (loading) return <h3>Loading team...</h3>;

  return (
    <>
      <PlayerDetail player={player} showDetail />
    </>
  );
}
