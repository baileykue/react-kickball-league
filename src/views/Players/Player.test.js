import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Player from './Player';

test("should render a player's details", async () => {
  const { container } = render(
    <MemoryRouter>
      <Player match={{ params: { id: 2 } }} />
    </MemoryRouter>
  );
  await screen.findByText('Hans Olo');
  expect(container).toMatchSnapshot();
});
