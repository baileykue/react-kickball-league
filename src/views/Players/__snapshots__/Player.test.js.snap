// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a player's details 1`] = `
<div>
  <div
    class="detail"
  >
    <div
      class="detail"
    >
      <h1>
        Hans Olo
      </h1>
      <h3>
        Position: 
        Catcher
      </h3>
      <label>
        Team:
        <a
          href="/teams/1"
        >
          Bridge City Sneakers
        </a>
      </label>
    </div>
  </div>
</div>
`;
